runtime/interpreter/interpreter_expression.go

```diff
@@ -523,6 +497,83 @@ func (interpreter *Interpreter) testEqual(left, right Value, expression *ast.Bin
 	)
 }
 
+func (interpreter *Interpreter) testComparison(left, right Value, expression *ast.BinaryExpression) BoolValue {
+	left = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Left,
+		},
+		left,
+	)
+
+	right = interpreter.Unbox(
+		LocationRange{
+			Location:    interpreter.Location,
+			HasPosition: expression.Right,
+		},
+		right,
+	)
```

Comment:
As mentioned in https://github.com/onflow/cadence/issues/2276#issuecomment-1496280606, we should probably keep the current behaviour of not automatically unboxing for non-equality comparisons

Reply:
Ok, I've removed the unboxing logic. Please let me know if there is anything else that needs to be done for this.